keps/sig-node/4188-kubelet-pod-readiness-api/README.md

```diff
@@ -66,76 +70,87 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
+  enables lean, efficient queries, but the API can also provide the full pod object when needed.
+  Because the full pod spec may contain secrets or other sensitive data, access must be tightly controlled.
 
 | Impact | Description|
 | ------- | ------------ |
-| + | Reliability - for various reasons kube-API might not be available but this doesn’t mean that local workloads are not accessible and on node system workloads should have the most recent data about pod's readiness even when kube-API is unreachable. |
-| + | Scalability - Reduce the load on kube-API by reducing the number of watchers and using Kubelet to fetch local Pods readiness. Fetching only Pods limited to one node is costly operation for kube-API. |
+| + | Reliability - for various reasons kube-API might not be available but this doesn’t mean that local workloads are not accessible and on node system workloads should have the most recent data about pod's state even when kube-API is unreachable. |
+| + | Scalability - Reduce the load on kube-API by reducing the number of watchers and using Kubelet to fetch local Pods information. Fetching only Pods limited to one node is costly operation for kube-API. |
 | + | Safety - Read-only API will not add security risks. |
-| + | Reduce resource consumption by workload. Using Kube-API we can fetch objects like PodSpec or PodStatus, for some on-node workload this might be unnecessary, with this API workload can reduce the resource consumption. |
+| + | Reduce resource consumption by workload. Using Kube-API we can fetch objects like PodSpec or PodStatus, for some on-node workload this might be unnecessary, with this API workload can reduce the resource consumption by requesting only the fields they need. The API can also provide the full pod spec for advanced use cases, but this is restricted to authorized local users. |
 | - | This API will add load to Kubelet (mitigation: API will be rate limited) |
-| - | Kubelet does not support RBAC authorization for gRPC. (mitigation: This API is designed to be accessible for all workloads running on the node without the authorization. The unix socket will be used for the connection and all exposed data will be carefully reviewed.) |
-| - | Limiting the scope of the API to the readiness information because we are not introducing the RBAC for this API. |
+| - | Kubelet does not support RBAC authorization for gRPC. (mitigation: This API is designed to be accessible only to local admin users. The unix socket will be secured with file permissions to restrict access to privileged users, and all exposed data will be carefully reviewed.) |
 
 ### Goals
 
-The goal of this API is to expose Pod readiness information directly from the
+- Expand Kubelet API Scope: Expand the Kubelet Pod Readiness API from readiness-specific to a more general pod information API, equivalent to the apiserver.
+- Support fieldmask-based filtering for lean responses.
```

Comment:
```suggestion
- Support fieldmask-based filtering for lean responses.
```

Can we declare it as a goal for the follow up KEP? Will returning the full Pod work in most cases?

The reason I suggest this as I do not see an API or config proposal for this in this KEP. And I am not very interested in solving the problem of coordinating requests from different API consumers and what their expectations will be